torch/csrc/dynamo/extra_state.cpp

```diff
@@ -114,18 +125,31 @@ void lookup(
     PyObject* f_locals,
     PyObject* backend,
     PyObject** maybe_cached_code,
-    const char** trace_annotation) {
+    const char** trace_annotation,
+    bool run_diff_guards) {
   size_t index = 0;
   CacheEntry* found = nullptr;
   py::handle locals(f_locals);
+  if (run_diff_guards && extra_state->cache_entry_list.size() > 1) {
+    // Move the first entry to the back. This is NOT a perf optimization, this
+    // is a necessity. It is possible that in the case of multiple cache
+    // entries, the first cache entry (which would be the latest recompile) does
+    // not have any guard managers with fail_count > 0. So, we just move it to
+    // the end.
+    extra_state->move_front_to_back();
```

Comment:
@jansel can you remind me why we attempt the most recently compiled cache entry as opposed to the oldest cache entry lol. I guess I can imagine some reasons. Reordering guards seems very dangerous in the unsafe eval world......

Reply:
MRU is useful for automatic dynamic

Reply:
The reordering logic is to "fail fast" where we try to check things that fail often first.  This is to mitigate the fact that we check multiple cache entries one-by-one.

I don't exactly understand why we are reordering this here though...